src/patient/PatientFiles.js

```diff
@@ -7,119 +7,127 @@ import TableHead from '@mui/material/TableHead';
-      </TableRow>
-    ))
-    || (error && <TableRow><TableCell><Error message={error} /></TableCell></TableRow>)
-    || <TableRow><TableCell><CustomSpinner /></TableCell></TableRow>
-  }
+        (data && (data.record_type === sortTag || sortTag === '') && search(query, data) && (
+          <>
+            <TableRow
+              hover
+              onClick={handleOpen}
+              sx={{ '&:last-child td, &:last-child th': { border: 0 }, textDecoration: 'none' }}
+            >
+              <TableCell component="th" scope="row">
+                {data.file_name}
+              </TableCell>
+              <TableCell component="th" scope="row">
+                {data.record_type}
+              </TableCell>
+              <TableCell>{data.uploader_name}</TableCell>
+              <TableCell />
+              <TableCell>{data.date_uploaded ?? ''}</TableCell>
+            </TableRow>
+            <FileModal handleOpen={handleOpen} handleClose={handleClose} open={open} data={data} />
+          </>
+        ))
+        || (error && <TableRow><TableCell><Error message={error} /></TableCell></TableRow>)
+        || (
+          <TableRow style={{ height: '10vh' }}>
+            <TableCell align="center" />
+            <TableCell align="center" />
+            <TableCell align="left">{`No ${sortTag} Files`}</TableCell>
+            <TableCell align="center" />
+          </TableRow>
+
+        )
+      }
     </>
   );
 };
 
 const PatientFiles = (props) => {
   const { data } = useGetPatientInfoQuery();
+  const [sortTag, setSortTag] = useState('');
+  const [searchTerm, setSearchTerm] = useState('');
 
-  const tableContent = () => {
+  const tableContent = (filter, query) => {
     if (data) {
       // using index for id for now
-      return data.files.map((fileHash) => <PatientFile fileHash={fileHash} key={fileHash} />);
+      return data.files.map((fileHash) => <PatientFile query={query} sortTag={filter} fileHash={fileHash} key={fileHash} />);
     } else {
       // Temporary, will be replaced with an error component or not
       return <TableRow><TableCell><CustomSpinner /></TableCell></TableRow>;
     }
   };
 
+  const onQueryChange = (query) => {
+    console.log(query);
+    setSearchTerm(query);
+  };
+
+  const handleSort = (filter) => {
+    setSortTag(filter);
+  };
+
   return (
     <div>
       <Box sx={{ display: 'flex', justifyContent: 'space-between' }}>
         <Typography variant="h1">Files</Typography>
-        <SearchBar />
+        <SearchBar onQueryChange={onQueryChange} />
       </Box>
-
+      <div>
+        <Button value="Medical history" onClick={(e) => handleSort(e.target.value)}>Medical History</Button>
+        <Button value="Blood test" onClick={(e) => handleSort(e.target.value)}>Blood Tests</Button>
+        <Button value="Procedure" onClick={(e) => handleSort(e.target.value)}>Procedures</Button>
+        <Button value="" onClick={(e) => handleSort(e.target.value)}>Reset</Button>
```

Comment:
FIX: onClick={(e) => { handleSort(e.target.value);  }} is preferred. Omitting the braces means this the return value of the function.